src/Symfony/Component/Validator/Constraints/LuhnValidator.php

```diff
@@ -60,23 +60,23 @@ public function validate(mixed $value, Constraint $constraint)
         $checkSum = 0;
         $length = \strlen($value);
 
-        // Starting with the last digit and walking left, add every second
-        // digit to the check sum
-        // e.g. 7  9  9  2  7  3  9  8  7  1  3
-        //      ^     ^     ^     ^     ^     ^
-        //    = 7  +  9  +  7  +  9  +  7  +  3
-        for ($i = $length - 1; $i >= 0; $i -= 2) {
-            $checkSum += $value[$i];
-        }
-
-        // Starting with the second last digit and walking left, double every
-        // second digit and add it to the check sum
-        // For doubles greater than 9, sum the individual digits
-        // e.g. 7  9  9  2  7  3  9  8  7  1  3
-        //         ^     ^     ^     ^     ^
-        //    =    1+8 + 4  +  6  +  1+6 + 2
-        for ($i = $length - 2; $i >= 0; $i -= 2) {
-            $checkSum += array_sum(str_split((int) $value[$i] * 2));
+        for ($i = $length - 1; $i >= 0; --$i) {
```

Comment:
```php
$i = $length;
while ($i--) {...}
```

Reply:
I'd keep here the loop, because otherwise this doesn't look better IMHO:
```
$i = $length;
while(--$i >= 0) {
```